src/legacy/core_plugins/metrics/public/kbn_vis_type.js

```diff
@@ -17,24 +17,22 @@
  * under the License.
  */
 
+import chrome from 'ui/chrome';
 import { MetricsRequestHandlerProvider } from './request_handler';
 import { i18n } from '@kbn/i18n';
 import { ReactEditorControllerProvider } from './editor_controller';
-import { VisFactoryProvider } from 'ui/vis/vis_factory';
 import { defaultFeedbackMessage } from 'ui/vis/default_feedback_message';
+import { visFactory } from '../../visualizations/public';
 
-import { PANEL_TYPES } from '../../common/panel_types';
+import { PANEL_TYPES } from '../common/panel_types';
 
-// register the provider with the visTypes registry so that other know it exists
-import { VisTypesRegistryProvider } from 'ui/registry/vis_types';
-VisTypesRegistryProvider.register(MetricsVisProvider);
+const uiSettings = chrome.getUiSettingsClient();
 
-export function MetricsVisProvider(Private) {
-  const VisFactory = Private(VisFactoryProvider);
-  const ReactEditorController = Private(ReactEditorControllerProvider).handler;
-  const metricsRequestHandler = Private(MetricsRequestHandlerProvider).handler;
+export function MetricsVis() {
```

Comment:
you can even get rid of this line, and just export visualization on line 35.

Reply:
Done